contracts/Aavegotchi/libraries/LibAppStorage.sol

```diff
@@ -366,6 +382,23 @@ struct AppStorage {
     // Auxiliary structs for Items Roles Registry
     // gotchiId => equippedDepositsInfo
     mapping(uint256 => GotchiEquippedDepositsInfo) gotchiEquippedDepositsInfo;
+
+  
+    // Parcel Roles
+
+    mapping(address => mapping(uint256 => mapping(bytes32 => RoleData )))  roles;
+
+    // tokenAddress => tokenId => owner
+    mapping(address => mapping(uint256 => address))  originalOwners;
+
+     // owner => tokenAddress => operator => isApproved
+    mapping(address => mapping(address => mapping(address => bool)))  tokenApprovals;
+
+    mapping(bytes32 => bool)  isRoleAllowed;
+
+    bytes32[]  allowedRoles;
+
```

Comment:
Each tokenAddress should have its own list of allowed roles
```suggestion
    // tokenAddress => roleId => isAllowed
    mapping(address => mapping(bytes32 => bool)) isRoleAllowed;

    bytes32[] allowedRoles;

```

Reply:
I need to create a function  function setAlloweRoles(address _tokenAddress, uint256 _tokenId, bytes32[] calldata _roles)

because as we changed the mapping isRoleAllowed to have the address we can't call in the constructor. 

Is it right ?